iroha/src/sumeragi.rs

```diff
@@ -35,18 +35,20 @@ pub enum Role {
     ProxyTail,
 }
 
-#[allow(dead_code)]
 impl Sumeragi {
     pub fn new(
         public_key: PublicKey,
         private_key: PrivateKey,
         peers: &[PeerId],
         peer_id: PeerId,
+        //TODO: get previos block hash from kura
         prev_block_hash: Option<Hash>,
         max_faults: usize,
         kura: Arc<Mutex<Kura>>,
     ) -> Result<Self, String> {
-        //TODO: check that that trusted peers contains this peer
+        if !peers.contains(&peer_id) {
+            return Err("Peers list should contain this peer.".to_string());
```

Comment:
Maybe we can push peer_id into peers in such a case?

Reply:
I also had this idea. But if you remember we load trusted peers from the json file and blockchain, so this list of peers will definitely include self.

Reply:
@takemiyamakoto Will we have `self` peer in a list of it's trusted peers?

Reply:
If we don't have self in the trusted peers list, then it would mean we would need to generate a custom file for every node. It's okay I guess, but not very elegant to my mind.